### DISCUSSION TITLE
UI Enhancements

### DISCUSSION THREAD

**Discussion by mrjones2014** [olimorris/codecompanion.nvim#33] [Ideas]
Idk if you've seen this but we could do some really freaking cool shit in this plugin using this 
https://nui-components.grapp.dev/

**Comment by olimorris** [comment]
Really, really cool!

I'd be interested to hear your ideas of where could improve the UI of the plugin.

**Reply by mrjones2014** [reply]
A chat buffer as like a pinned sidebar is one idea I had. It would also make it easy to add loading indicators to the chat buffers. You could also make the settings at the top of the chat buffer into form fields.

maybe I’ll play around with it this weekend to see what I can put together.

**Reply by olimorris** [reply]
Okay, that would be really awesome! And actually could add a lot of new possibilities to the plugin after thinking about it.

I've wanted to be able to add support for files or images, to leverage the vision capabilities of some of the models, but I've been unsure of how to do incorporate this nicely into the chat buffer. Perhaps Treesitter to parse some markdown tags (but feels like it would get messy). So these components could be a great fit!

**Comment by lazymaniac** [comment]
I have other idea related to UI enhancements. Something like prompt builder. Let's say user can define a template for the promt:

```mardown
I have the following SQL table defintition:
%[SQL Table:]
I want you to: 
%[Query:]
```

and based on prvided tags some ui form can be constructed and displayed when user wants to run this action.

```mardown
---------------------------
SQL Query Action
---------------------------
SQL Table: 
---------------------------
Fill text

---------------------------
Query:
---------------------------
Fill text

---------------------------
Result: 
---------------------------
I have the following SQL table defintition:
Fill text
I want you to: 
Fill text
```

**Comment by olimorris** [comment]
Discussion moved to #495
